src/main/java/com/commercetools/sync/internals/utils/UnorderedCollectionSyncUtils.java

```diff
@@ -0,0 +1,60 @@
+package com.commercetools.sync.internals.utils;
+
+import io.sphere.sdk.commands.UpdateAction;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+import static com.commercetools.sync.commons.utils.CollectionUtils.emptyIfNull;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * This class is only meant for the internal use of the commercetools-sync-java library.
+ */
+public final class UnorderedCollectionSyncUtils {
+
+    /**
+     * Compares a list of {@code newDrafts} with a map of {@code oldResourcesMap} and for every missing matching draft
+     * in the {@code oldResourcesMap}: a remove update action is created using the {@code removeUpdateActionMapper}.
+     * The final result is a list of all the remove update actions.
+     *
+     * @param oldResourcesMap          a map that consists of entries where each entry has a key=[resource matcher] and
+     *                                 the value=[the old resource itself].
+     * @param newDrafts                a list of the new drafts to compare to the old collection.
+     * @param keyMapper                a function that uses the draft to get its key matcher.
+     * @param removeUpdateActionMapper a function that uses the old resource to build a remove update action.
+     * @param <T>                      type of the resulting update actions.
+     * @param <S>                      type of the new resource key.
+     * @param <U>                      type of the old resource.
+     * @param <V>                      type of the new resource.
+     * @return a list of all the remove update actions. If there are no missing matching drafts, an empty list is
+     *         returned.
+     */
+    @Nonnull
+    public static <T, S, U, V> List<UpdateAction<T>> buildRemoveUpdateActions(
+        @Nonnull final Map<S, U> oldResourcesMap,
```

Comment:
This is particular opinion but I think would be better to pass the List of oldResources rather than a Map. I would create the map inside this function based on the list. why?

- The map is a "processed" parameter (the list of old resources is the original input) so it creates a dependency with the function that transforms the list to map.
- Easy to test and maintain because the code is more decoupled and we don't need to create this map in our unit tests
- The performance is not an issue even if we generate the map inside each buildUpdateAction (remove, add, etc...) since the cost is O(n).

Reply:
agreed!: 5bb3cc4